app/components/app_consent_component.rb

```diff
@@ -55,20 +55,36 @@ def contact_parent_or_guardian_link(consents)
 
   def consents_grouped_by_parent
     @consents_grouped_by_parent ||=
-      @patient_session.consents.group_by { |consent| summary_for(consent) }
+      @patient_session.consents.group_by(&:summary_with_consenter)
   end
 
   private
 
-  def summary_for(consent)
-    summary =
-      if consent.respond_to?(:response_not_provided?) &&
-           consent.response_not_provided?
-        "Contacted #{consent.name}"
-      else
-        "#{consent.human_enum_name(:response).capitalize} by #{consent.name}"
-      end
-    summary += " (#{consent.who_responded})" unless consent.via_self_consent?
-    summary
+  def render_grouped_consents(consents)
+    first_consent = consents.first
+    first_refused_consent = consents.find_all(&:response_refused?).first
```

Comment:
I think a plain old `find` would work here too.

```suggestion
    first_refused_consent = consents.find(&:response_refused?)
```

Reply:
Yep, updated in [b7a8b34](https://github.com/nhsuk/manage-childrens-vaccinations/pull/1206/commits/b7a8b348e2c1ecc34071fea5aa42f8b38b19eb67)